cf-serverd/server_transform.c

```diff
@@ -687,6 +687,26 @@ static PromiseResult KeepServerPromise(EvalContext *ctx, const Promise *pp, ARG_
         return VerifyClassPromise(ctx, pp, NULL);
     }
 
+    /* Warn if promise locking was used with a promise that doesn't support it.
+     * That applies to both of the below promise types while 'vars' and
+     * 'classes' handle this on their own. */
+    int ifelapsed = PromiseGetConstraintAsInt(ctx, "ifelapsed", pp);
+    if (ifelapsed != CF_NOINT)
+    {
+        Log(LOG_LEVEL_WARNING, "ifelapsed attribute specified in action "
+            "body for %s promise '%s', but %s promises do not support "
+            "promise locking.", pp->parent_promise_type->name,
+            pp->promiser, pp->parent_promise_type->name);
```

Comment:
Do not break string literals, they harm grepability of the code. Try to break on punctuation or % qualifiers. I would write this one as:

```
Log(LOG_LEVEL_WARNING, 
    "ifelapsed attribute specified in action body for %s promise '%s',"
    " but %s promises do not support promise locking", 
    pp->parent_promise_type->name, pp->promiser, pp->parent_promise_type->name);
```

It's ok if string literals go a bit over 78 column. 

Also do not end log messages with punctuation.